iocore/eventsystem/UnixEventProcessor.cc

```diff
@@ -477,6 +477,12 @@ EventProcessor::spawn_event_threads(EventType ev_type, int n_threads, size_t sta
 void
 EventProcessor::initThreadState(EThread *t)
 {
+  // #10129: Initialize DbgCtl early on so that we process the PCRE regex before
+  // we load plugins. Otherwise, PCRE compilation on first use will try to grab
+  // the dl_load_lock, which might race and deadlock with our dlopen calls for
+  // plugins.
+  DbgCtl::update();
```

Comment:
I don't see how this would do any good.  This function doesn't result in a call to Diags::tag_actived(), while the Registry mutex is unlocked.

Reply:
The idea is that it will initialize the thread_local PCRE structure which acquires the dl_load_lock. If we do that early on, before plugins are loaded, then we avoid the race condition for the threads that initialize themselves through this function.